teps/0103-skipping-reason.md

```diff
@@ -0,0 +1,215 @@
+	
+	// None means the task was not skipped
+	None SkippingReason = "None"
+	
+)
+```
+
+### Example
+
+Take the example below where "skip-this-task" `PipelineTask` is skipped because of its `when`
+expressions evaluating to false:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: pipelinerun-skip-task-run-task
+spec:
+  pipelineSpec:
+    tasks:
+      - name: skip-this-task
+        when:
+          - input: foo
+            operator: in
+            values:
+              - bar
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: exit 1
+      - name: run-this-task
+        when:
+          - input: foo
+            operator: in
+            values:
+              - foo
+        taskSpec:
+          steps:
+            - name: echo
+              image: ubuntu
+              script: "echo run this task"
+```
+
+When the `Pipeline` above is executed, `reason` field would be added to the `skippedTasks`
+section indicating that "skip-this-task" `PipelineTask` was skipped because of its `when`
+expressions. 
+
+```yaml
+status:
+  completionTime: "2022-04-06T10:21:56Z"
+  conditions:
+    - lastTransitionTime: "2022-04-06T10:21:56Z"
+      message: 'Tasks Completed: 1 (Failed: 0, Cancelled 0), Skipped: 1'
+      reason: Completed
+      status: "True"
+      type: Succeeded
+  ...
+  skippedTasks:
+    - name: skip-this-task
+      reason: "WhenExpressionsSkip"
+      whenExpressions:
+        - input: foo
+          operator: in
+          values:
+            - bar
+  ...
+  taskRuns:
+    pipelinerun-skip-task-run-task-run-this-task:
+      pipelineTaskName: run-this-task
+      status:
+        completionTime: "2022-04-06T10:21:56Z"
+        conditions:
+          - lastTransitionTime: "2022-04-06T10:21:56Z"
+            message: All Steps have completed executing
+            reason: Succeeded
+            status: "True"
+            type: Succeeded
+        podName: pipelinerun-skip-task-run-task-run-this-task-pod
+        ...
+```
+
+## References
+
+* Tekton Enhancement Proposals:
+  * [TEP-0007: Conditions Beta][tep-0007]
+  * [TEP-0059: Skipping Strategies][tep-0059]
+* Issues:
+  * [Tekton Pipelines Issue 4738: Skipping Reason][issue-4738]
+  * [Tekton Pipelines Issue 4571: Task skipped when parallel task fails][issue-4571]
+  * [Tekton Community Slack Thread][slack]
+
+[tep-0007]: https://github.com/tektoncd/community/blob/main/teps/0007-conditions-beta.md
+[tep-0059]: https://github.com/tektoncd/community/blob/main/teps/0059-skipping-strategies.md
+[skipped-tasks]: https://github.com/tektoncd/pipeline/blob/053833cb10f3829d5a366daa1f431b293dcf3285/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L466-L476
+[issue-4738]: https://github.com/tektoncd/pipeline/issues/4738
+[issue-4571]: https://github.com/tektoncd/pipeline/issues/4571
+[slack]: https://tektoncd.slack.com/archives/CK3HBG7CM/p1642349040014100
```

Comment:
just fyi I don't think this slack link is available any longer, do you have a screenshot or text snippet of what was said? if not it's not a big deal

Reply:
we lose threads because we don't have a premium account so it's capped at 10K messages - added an item to the community meeting to discuss getting a premium subscription because it's our main means of communication

unfortunately, I didn't take a screenshot of that particular thread - hopefully we can recover it and others if we get a paid subscription 🤞🏾